src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,50 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        LocalDate startDate = LocalDate.parse(dateFrom, formatter);
+        LocalDate endDate = LocalDate.parse(dateTo, formatter);
+        LocalDate[] datesFromArray = new LocalDate[data.length];
+        String[] namesFromArray = new String[data.length];
+        String[] hoursFromArray = new String[data.length];
+        String[] salaryFromArray = new String[data.length];
+        int[] salarys = new int[data.length];
+        int[] salary = new int[names.length];
+        String line = System.lineSeparator();
```

Comment:
these lines 12-18 are redundant. Before the loop, you only need a variable StringBuilder. Then, in the outer loop iterate through names arr and calculate salary for each name. In the inner loop go through each row in data and check ifs data suits us and the name is the same as current (names[i]). That will make the code shorter